version_test.go

```diff
@@ -23,10 +23,11 @@ import (
 	"go.opentelemetry.io/otel"
 )
 
-// regex taken from https://github.com/Masterminds/semver/tree/v3.1.1
-var versionRegex = regexp.MustCompile(`^v?([0-9]+)(\.[0-9]+)?(\.[0-9]+)?` +
-	`(-([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?` +
-	`(\+([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?$`)
+// regex taken from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+var versionRegex = regexp.MustCompile(`^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)` +
```

Comment:
@pellared I have also changed regex string here, is this okay?

Reply:
yup 👍